File: old/speed.c
#include <time.h>
#include <stdio.h>
#include <stdlib.h>

int *insertion_sort(int *arr, int n) {
    int i, j, key;
    for (i = 1; i < n; i++) {
        key = arr[i];
        j = i - 1;
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = key;
    }
    return arr;
}

void _wait(int n) {
    for (; n > 0; n--);
}

int main(int argc, char *argv[]) {
    srand(time(NULL));
    if (argc == 2) {
        int SIZE = atoi(argv[1]);
        int *arr = malloc(SIZE * sizeof(int));
        for (int i = 0; i < SIZE; i++) {
            arr[i] = rand() % 1000;
        }
        //_wait(SIZE);
        clock_t start = clock();
        insertion_sort(arr, SIZE);
        clock_t end = clock();
        printf("Time taken: %f seconds.\n", (double)(end - start) / CLOCKS_PER_SEC);
        return 0;
    }
    printf("Usage: ./speed <size of array>\n");
    return 1;
}

File: old/speed.js
let SIZE = process.argv[2];

// insertion sort of an array of integers
function insertionSort(arr) {
    for (let i = 1; i < arr.length; i++) {
        let j = i - 1;
        let tmp = arr[i];
        while (j >= 0 && (arr[j] > tmp)) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = tmp;
    }
    return arr;
}

function wait(n) {
    for (let i = 0; i < n; i++);
}

// populate random array of size SIZE
let arr = [];
for (let i = 0; i < SIZE; i++) {
    arr.push(Math.floor(Math.random() * 1000));
}

// time
// wait(SIZE);
let start = new Date().getTime();
insertionSort(arr);
let end = new Date().getTime();
let time = (end - start) / 1000;
console.log('Time taken: ' + time + ' seconds.');